tests/service/data-util.service.spec.ts

```diff
@@ -20,51 +20,104 @@ import { TestBed, inject } from '@angular/core/testing';
+        expect(service.abbreviate('Hello Jhipster lets test the data utils function')).toBe('Hello Jhipster ...s function');
+    }));
 
-        it('should abbreviate the text and append +++', inject([JhiDataUtils], (service: JhiDataUtils) => {
-            expect(service.abbreviate('Hello Jhipster lets test the data utils function', '+++')).toBe('Hello Jhipster +++s function');
-        }));
+    it('should abbreviate the text and append +++', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.abbreviate('Hello Jhipster lets test the data utils function', '+++')).toBe('Hello Jhipster +++s function');
+    }));
 
-        it('should return the bytesize of the text', inject([JhiDataUtils], (service: JhiDataUtils) => {
-            expect(service.byteSize('Hello Jhipster')).toBe(`10.5 bytes`);
-        }));
+    it('should return the bytesize of the text', inject([JhiDataUtils], (service: JhiDataUtils) => {
+        expect(service.byteSize('Hello Jhipster')).toBe(`10.5 bytes`);
+    }));
 
-        it('should download the csv file', inject([JhiDataUtils], (service: JhiDataUtils) => {
+    it('should download the csv file', inject([JhiDataUtils], (service: JhiDataUtils) => {
             const tempLink = document.createElement('a');
             jest.spyOn(tempLink, 'click');
             jest.spyOn(document, 'createElement').mockReturnValue(tempLink);
-            // call downloadFile function
-            // csv content:
-            // ID,Name
-            // 1,Toto
-            const contentType = 'text/csv';
-            const data = 'SUQsTmFtZQ0KMSxUb3Rv';
-            const fileName = 'test-download-file.csv';
-            service.downloadFile(contentType, data, fileName);
-            expect(document.createElement).toHaveBeenCalledTimes(1);
-            expect(document.createElement).toHaveBeenCalledWith('a');
+        // call downloadFile function
+        // csv content:
+        // ID,Name
+        // 1,Toto
+        const contentType = 'text/csv';
+        const data = 'SUQsTmFtZQ0KMSxUb3Rv';
+        const fileName = 'test-download-file.csv';
+        service.downloadFile(contentType, data, fileName);
+        expect(document.createElement).toHaveBeenCalledTimes(1);
+        expect(document.createElement).toHaveBeenCalledWith('a');
             expect(tempLink.target).toBe('_blank');
             expect(tempLink.download).toBe('test-download-file.csv');
             expect(tempLink.click).toHaveBeenCalledTimes(1);
             expect(tempLink.click).toHaveBeenCalledWith();
-        }));
+    }));
+
+    it('should execute the toBase64()', inject([JhiDataUtils], (service: JhiDataUtils) => {
+
+        spyOn(service, 'toBase64');
+
+        const eventSake = {
+            target: {
+                files: [{}]
+            }
+        };
+
+        service.setFileData(eventSake, null, null, false);
+
+        setTimeout(() => {
```

Comment:
timeouts in tests doesn't looks good. can you try angular async function ?

Reply:
I agree. It was the only way I knew how.
I'll update the PR

Reply:
Take a look at this with Jest, should do the job: https://jestjs.io/docs/en/timer-mocks

Reply:
IMO, timer mocks wouldn't be useful as underlying code doesn't uses any timer method. Jest is good, but, angular also provides `fakeAsync` and `tick` to handle timer scenarios in testing.